GoogleForCreators/web-stories-wp
# Fix layer re-arrangement inside groups using shortcuts

## Context
Layer arrangement using shortcuts got broken after adding the groups feature. This PR aims to fix it.


## Summary
Allows moving a layer up/down considering groups. A layer is moved inside/outside of the groups depending on the position and grouping.

If the layer is the first item of the group, and UP is pressed, the layer is moved outside of that group, the position stays the same. If the layer is the last item of the group and DOWN is pressed, the layer is moved outside of the group, the position stays the same.

The same logic is also used for moving a layer into a group.

For other cases, the logic remains as currently is.


## Relevant Technical Choices



## To-do



## User-facing changes
Groups are now added / removed from layers when using shortcuts.


## Testing Instructions






This PR can be tested by following these steps:

1. Add 3 elements and group the first 2 elements.
2. Now select the ungrouped element.
3. Use <kbd>cmd</kbd>+<kbd>up</kbd> and <kbd>cmd</kbd>+<kbd>down</kbd> to change the position of the layer.
4. Look at the Layer Panel at the same time -- verify that the selected layer is moving as expected into a group and out of the group when moving it up and down in the layers list.

![groups](https://user-images.githubusercontent.com/3294597/177756755-2b24fc45-9271-4cbc-8f48-08e15f20f7d6.gif)


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11784
